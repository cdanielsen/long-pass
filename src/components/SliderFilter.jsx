import React from 'react'
import PropTypes from 'prop-types'

function SliderFilter (props) {
  return (
    <div>
      <h2>{props.title}</h2>
      <div>
        <input
          id={props.id}
          type="range"
          min={props.min}
          max={props.max}
          step='1'
          onChange={props.newSelectionHandler}
          value={props.value}>
        </input>
        <h2>{props.value}</h2>
      </div>
    </div>
  )
}

SliderFilter.propTypes = {
  title: PropTypes.string,
  id: PropTypes.string,
  min: PropTypes.number,
  max: PropTypes.number,
  newSelectionHandler: PropTypes.func.isRequired,
  value: PropTypes.number.isRequired
}

export default SliderFilter
