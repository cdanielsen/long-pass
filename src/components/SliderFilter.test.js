import React from 'react'
import SliderFilter from './SliderFilter.jsx'
import { shallow } from 'enzyme'

describe('Slider Filter component', () => {
  const props = {
    title: 'Sweet sweet slider',
    id: '1',
    min: 1,
    max: 10,
    value: 5,
    newSelectionHandler: jest.fn()
  }

  test('Given props, should return with the correct attributes', () => {
    const wrapper = shallow(<SliderFilter {...props} />)
    const inputElementProps = wrapper.find('input').props()

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('h3').first().text()).toEqual(props.title)
    expect(inputElementProps.id).toEqual(props.id)
    expect(inputElementProps.onChange).toEqual(props.newSelectionHandler)
    expect(inputElementProps.min).toEqual(props.min)
    expect(inputElementProps.max).toEqual(props.max)
    expect(inputElementProps.value).toEqual(props.value)
  })

  test('When the input value changes, onChange handler should be called', () => {
    const mockEvent = {target: {value: 3}}
    const wrapper = shallow(<SliderFilter {...props} />)

    wrapper.find('input').simulate('change', mockEvent)
    expect(props.newSelectionHandler.mock.calls.length).toEqual(1)
    expect(props.newSelectionHandler).toHaveBeenCalledWith(mockEvent)

    props.newSelectionHandler.mockReset()
  })
})
