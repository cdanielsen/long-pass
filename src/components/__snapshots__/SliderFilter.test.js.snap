// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderFilter component Given props, should return with the correct attributes 1`] = `
<div>
  <h3>
    Sweet sweet slider
  </h3>
  <input
    id="1"
    max={10}
    min={1}
    onChange={[Function]}
    step="1"
    type="range"
    value={5}
  />
  <h3>
    5
  </h3>
</div>
`;
